File: Twosum.cpp
#include<bits/stdc++.h>

using namespace std;



//  g++ -o Twosum.exe Twosum.cpp && ./Twosum.exe




int main(){

    vector<int> nums = {3,2,4};

    int n = nums.size();
    int target = 6;

    unordered_map<int, int> N;

    for(int i=0;i<n;i++){

        int sub = target - nums[i];   // ith - target 

        if(N.count(sub)){

           cout<<"["<<N[sub]<<","<<i<<"]"<<endl;       
        }

        N[nums[i]] = i;

    }

    
  return 0;
}

File: Python/Twosum.py
from typing import List


def twosum(nums: List[int], target=int) -> List[int]:
    rel = {}
    for i, n in enumerate(nums):
        sub = target - i
        if sub in rel:
            return [rel[sub], i]
        rel[n] = i


nums = [2, 7, 11, 15]
target = 9

result = twosum(nums, target)
print(result)
